extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -441,15 +444,24 @@ export class AmpStoryPage extends AMP.BaseElement {
 
   /** @override */
   layoutCallback() {
+    this.waitForMediaLayout_().then(
+      () => {
+        this.pageLoadResolveFn_(this.markPageAsLoaded_());
+      },
+      reason => {
+        this.pageLoadRejectFn_(reason);
+      }
+    );
+
     upgradeBackgroundAudio(this.element);
     this.muteAllMedia();
     this.getViewport().onResize(
       debounce(this.win, () => this.onResize_(), RESIZE_TIMEOUT_MS)
     );
     return Promise.all([
       this.beforeVisible(),
-      this.mediaLayoutPromise_,
       this.mediaPoolPromise_,
+      this.pageLoadPromise_,
```

Comment:
Actually, now that we're trying to delay the `layoutCallback` to resolve only when the page is fully loaded (which is consistent with what `amp-audio` and `amp-video` are doing, so :+1: ), we should be able to:
- remove our custom `whenLoaded` method, and rely on the `LOAD_END` runtime signal that's sent when `layoutCallback` completed
- this allows us to remove the three `this.pageLoad..._` ivars here, and make the code a bit easier and more AMP compliant :)

Cool PR!